pkg/agent/controller/endpointslice.go

```diff
@@ -0,0 +1,228 @@
+			},
+		},
+		&discovery.EndpointSlice{},
+		0,
+		cache.Indexers{},
+	)
+
+	e.endpointSliceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			klog.Infof("EndpointSlice %q added", key)
+			if err == nil {
+				e.endPointSlicequeue.Add(key)
+			}
+		},
+		UpdateFunc: func(obj interface{}, new interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(new)
+			klog.Infof("EndpointSlice %q updated", key)
+			if err == nil {
+				e.endPointSlicequeue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			klog.Infof("EndpointSlice %q deleted", key)
+			if err == nil {
+				var ok bool
+				var endPointSlice *discovery.EndpointSlice
+				if endPointSlice, ok = obj.(*discovery.EndpointSlice); !ok {
+					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						klog.Errorf("Failed to get deleted EndpointSlice object for key %s, serviceImport %v", key, endPointSlice)
+						return
+					}
+
+					endPointSlice, ok = tombstone.Obj.(*discovery.EndpointSlice)
+
+					if !ok {
+						klog.Errorf("Failed to convert deleted tombstone object %v  to endPointslice", tombstone.Obj)
+						return
+					}
+				}
+				e.endPointSlicequeue.Add(key)
+				e.endpointSliceDeletedMap.Store(endPointSlice, key)
+			}
+		},
+	})
+
+	go e.endpointSliceInformer.Run(e.stopCh)
+	go e.runEndpointSliceWorker(e.endpointSliceInformer, e.endPointSlicequeue)
+
+	go func(stopCh <-chan struct{}) {
+		<-stopCh
+		e.endPointSlicequeue.ShutDown()
+
+		klog.Infof("ServiceImport Controller stopped")
+	}(stopCh)
+
+	return nil
+}
+
+func (e *EndpointSliceController) runEndpointSliceWorker(informer cache.SharedIndexInformer, queue workqueue.RateLimitingInterface) {
+	for {
+		keyObj, shutdown := queue.Get()
+		if shutdown {
+			klog.Infof("Lighthouse watcher for ServiceImports stopped")
+			return
+		}
+
+		key := keyObj.(string)
+
+		func() {
+			defer queue.Done(key)
+			obj, exists, err := informer.GetIndexer().GetByKey(key)
+
+			if err != nil {
+				klog.Errorf("Error retrieving the object with key  %s from the cache: %v", key, err)
+				// requeue the item to work on later
+				queue.AddRateLimited(key)
+
+				return
+			}
+
+			if exists {
+				err = e.endPointSliceCreatedOrUpdated(obj, key)
+			} else {
+				err = e.endPointSliceDeleted(key)
+			}
+
+			if err != nil {
+				if !exists {
+					e.endpointSliceDeletedMap.Store(key, obj)
+				}
+
+				queue.AddRateLimited(key)
+			} else {
+				queue.Forget(key)
+			}
+		}()
+	}
+}
+
+func (e *EndpointSliceController) endPointSliceCreatedOrUpdated(obj interface{}, key string) error {
+	endPointSlice := obj.(*discovery.EndpointSlice)
+	labels := endPointSlice.GetObjectMeta().GetLabels()
+
+	if labels[labelSourceCluster] == e.clusterID {
+		klog.Infof("SOurce Label %s is same as %s", labels[labelSourceCluster], e.clusterID)
```

Comment:
```suggestion
		klog.Infof("Source cluster label %s matches - ignoring", labels[labelSourceCluster])
```

Should this be log.DEBUG or TRACE?

Reply:
Removed the controller.